### PR TITLE
added rollbar config

### PR DESCRIPTION
**Pull request by Chee-Ng**

### CODE REVIEW

**Comment by tommyb82** [comment]
The test suite is not passing as Spring cannot instantiate the `RollbarConfig` config class, because the env var `rollbar.access.token` is missing - suggest resolving that first.  Might be worth mocking the `Rollbar` instance in the tests otherwise (presumably) we are going to see exceptions as it tried to call out to the rollbar service with an incorrect access token.  Also, we probably don't want the test suite calling a real external service anyway...

File: src/main/java/uk/gov/crowncommercial/dts/scale/service/gm/config/RollbarConfig.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package uk.gov.crowncommercial.dts.scale.service.gm.config;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/service/gm/config/RollbarConfig.java]: Remove unused import

File: src/main/java/uk/gov/crowncommercial/dts/scale/service/gm/config/RollbarConfig.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/service/gm/config/RollbarConfig.java]: Remove unused import

File: src/main/java/uk/gov/crowncommercial/dts/scale/service/gm/controller/GlobalErrorHandler.java [hunk 3]
```diff
@@ -28,12 +33,16 @@
...
+	
+  @Autowired
+  private Rollbar rollbar;
 
   @ResponseStatus(HttpStatus.BAD_REQUEST)
   @ExceptionHandler({ValidationException.class, HttpMessageNotReadableException.class})
   public ApiErrors handleValidationException(final Exception exception) {
 
     log.trace("Request validation exception", exception);
+    rollbar.error(exception,"Request validation exception");
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/service/gm/controller/GlobalErrorHandler.java]: This should perhaps be logged at `INFO` level - it's not a system error as such, just catching client input validation - hence the `trace` level logging.  Also - **if** all log output is supposed to be redirected to rollbar, then I assume we should remove the console log statements?

File: src/main/java/uk/gov/crowncommercial/dts/scale/service/gm/controller/GlobalErrorHandler.java [hunk 4]
```diff
@@ -59,6 +69,7 @@ public ApiErrors handleValidationException(final Exception exception) {
   public ApiErrors handleResourceNotFoundException(final ResourceNotFoundException exception) {
 
     log.info("Requested resource not found", exception);
+    rollbar.error(exception, "Requested resource not found");
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/service/gm/controller/GlobalErrorHandler.java]: Again, it's not a system error really - this handler is just catching the case where a client requests something that doesn't exist.  Hence the 404 response code and `info` level logging

**Review by tommyb82 — COMMENTED** [review]
Few more suggestions / comments for consideration 👍

File: pom.xml [hunk 5]
```diff
@@ -69,6 +69,18 @@
             <optional>true</optional>
         </dependency>
 
+		<dependency>
+		  <groupId>com.rollbar</groupId>
```
Comment (tommyb82) [inline pom.xml]: Only the `rollbar-spring-boot-webmvc` is required - it includes this one transitively so no need for both

File: src/main/java/uk/gov/crowncommercial/dts/scale/service/gm/controller/GlobalErrorHandler.java [hunk 6]
```diff
@@ -94,6 +107,7 @@ public ApiErrors handleUnknownException(final Exception exception) {
     Object exception = request.getAttribute("javax.servlet.error.exception");
 
     log.error("Unknown container/filter exception", exception);
+    rollbar.error((Throwable) exception, "Unknown container/filter exception");
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/service/gm/controller/GlobalErrorHandler.java]: Might be better to do the cast to type `Throwable` on line 107, then it is explicitly the correct type for both log statements.

**Review by tommyb82 — APPROVED** [review]
Just those minor comments - looking good though 👍

File: src/main/java/uk/gov/crowncommercial/dts/scale/service/gm/controller/GlobalErrorHandler.java (hunk 6 above)
Comment (Chee-Ng) [reply src/main/java/uk/gov/crowncommercial/dts/scale/service/gm/controller/GlobalErrorHandler.java]: Is this something you are looking for 
`Throwable exception = (Throwable) request.getAttribute("javax.servlet.error.exception");`

Comment (tommyb82) [reply src/main/java/uk/gov/crowncommercial/dts/scale/service/gm/controller/GlobalErrorHandler.java]: Yep exactly

**Pull request merged**